tests/ci/pr_info.py

```diff
@@ -304,57 +304,66 @@ def has_changes_in_submodules(self):
         return False
 
     def can_skip_builds_and_use_version_from_master(self):
-        # TODO: See a broken loop
         if FORCE_TESTS_LABEL in self.labels:
             return False
 
         if self.changed_files is None or not self.changed_files:
             return False
 
-        for f in self.changed_files:
-            # TODO: this logic is broken, should be fixed before using
-            if (
-                not f.startswith("tests/queries")
-                or not f.startswith("tests/integration")
-                or not f.startswith("tests/performance")
-            ):
-                return False
+        return not any(
+            f.startswith("programs")
+            or f.startswith("src")
+            or f.startswith("base")
+            or f.startswith("cmake")
+            or f.startswith("rust")
+            or f == "CMakeLists.txt"
+            or f == "tests/ci/build_check.py"
+            for f in self.changed_files
+        )
 
-        return True
 
-    def can_skip_integration_tests(self):
-        # TODO: See a broken loop
-        if FORCE_TESTS_LABEL in self.labels:
-            return False
+def can_skip_integration_tests(self, versions: List[str]):
+    if FORCE_TESTS_LABEL in self.labels:
+        return False
 
-        if self.changed_files is None or not self.changed_files:
-            return False
+    # If docker image(s) relevant to integration tests are updated
+    if self.sha not in versions:
```

Comment:
`versions` in this case is `0-{sha}`, so we need to check `any(self.sha in version for version in versions)`